### PR TITLE
Infobox template controller

### PR DESCRIPTION
**Pull request by nikodamn**
* Improve Portable Infobox tests to push template and article content before executing the test
* Introduce helper class for loading text file's content to String

Reviewers:
@ludwikkazmierczak 

5.10.2016: This PR is blocked by staging env maintenance. Please **DON'T CLOSE** this PR.

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/common/core/helpers/FileTextLoader.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+public class FileTextLoader {
+
+  public String loadFileTextContent(String filename) {
+    StringBuilder textContent = new StringBuilder();
+    String NL = System.getProperty("line.separator");
+    Scanner scanner = null;
+    try {
+      scanner = new Scanner(new File(PageContent.TEXT_FILE_RESOURCES_PATH + filename), "UTF-8");
+    } catch (FileNotFoundException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/helpers/FileTextLoader.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/common/core/helpers/FileTextLoader.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      e.printStackTrace();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/helpers/FileTextLoader.java]: CRITICAL Use a logger to log this exception. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1148)

File: src/test/java/com/wikia/webdriver/common/core/helpers/FileTextLoader.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+
+public class FileTextLoader {
+
+  public String loadFileTextContent(String filename) {
+    StringBuilder textContent = new StringBuilder();
+    String NL = System.getProperty("line.separator");
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/helpers/FileTextLoader.java]: MAJOR Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00117)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues:
* CRITICAL 1 critical
* MAJOR 5 major

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/common/core/helpers/FileTextLoader.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import com.wikia.webdriver.common.contentpatterns.PageContent;
+import com.wikia.webdriver.common.logging.PageObjectLogging;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+
+public class FileTextLoader {
+
+  public String loadFileTextContent(String filename) {
```
Comment (ludwikkazmierczak) [inline src/test/java/com/wikia/webdriver/common/core/helpers/FileTextLoader.java]: Load WikiText Content?

File: src/test/java/com/wikia/webdriver/common/core/helpers/ContentLoader.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+import com.wikia.webdriver.common.logging.PageObjectLogging;
+
+/**
+ * Helper class to load content from external files
+ */
+
+public class ContentLoader {
+
+  private final static String TEXT_FILES_PATH_FORMAT = "TextFiles/%s";
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/helpers/ContentLoader.java]: MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AModifiersOrderCheck)

File: src/test/java/com/wikia/webdriver/common/core/helpers/ContentLoader.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+
+import com.wikia.webdriver.common.logging.PageObjectLogging;
+
+/**
+ * Helper class to load content from external files
+ */
+
+public class ContentLoader {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/helpers/ContentLoader.java]: MAJOR Add a private constructor to hide the implicit public one. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1118)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 1 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Comment by ludwikkazmierczak** [comment]
LGTM

**Pull request merged**